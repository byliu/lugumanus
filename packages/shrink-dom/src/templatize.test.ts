import { afterEach, beforeEach, describe, expect, test } from 'bun:test';
import path from 'node:path';
import { JSDOM } from 'jsdom';
import { DOMContentExtractor } from './dom-content-extractor';
import { DOMDomainAnalyzer } from './dom-domain-analyzer';
import { HTMLTemplateProcessor } from './templatize';

type Global = typeof globalThis & {
  window: Window;
  document: Document;
};

describe('DOMDomainAnalyzer', () => {
  let dom: JSDOM;
  let originalWindow: Global['window'] | undefined;
  let originalDocument: Global['document'] | undefined;

  beforeEach(async () => {
    // 保存原始的全局变量
    originalWindow = global.window as Global['window'] | undefined;
    originalDocument = global.document as Global['document'] | undefined;

    const html = await Bun.file(path.join(import.meta.dirname, './mock/mock2.html')).text();

    // 创建新的 JSDOM 实例
    dom = new JSDOM(html);

    // 设置全局变量
    global.window = dom.window as unknown as Window & typeof globalThis;
    global.document = dom.window.document;

    // 模拟 window.getComputedStyle
    global.window.getComputedStyle = (element: Element) => {
      const style = {} as CSSStyleDeclaration;

      // 为测试元素设置默认样式
      style.display = 'block';
      style.visibility = 'visible';
      style.opacity = '1';
      style.width = '100';
      style.height = '50';
      style.cursor = 'default';

      // 为特定元素设置特殊样式
      if (element.id === 'hidden-div') {
        style.display = 'none';
      }

      if (element.id === 'pointer-div') {
        style.cursor = 'pointer';
      }

      return style;
    };

    // 确保 Element.prototype 上有 style 属性的所有需要的方法
    if (typeof Element !== 'undefined' && Element.prototype) {
      // 扩展 Element.prototype.style 以支持所有需要的属性
      Object.defineProperty(Element.prototype, 'style', {
        get: function () {
          if (!this._style) {
            this._style = {
              filter: '',
              transition: '',
              setProperty: function (prop: string, value: string) {
                this[prop] = value;
              },
              getPropertyValue: function (prop: string) {
                return this[prop] || '';
              },
            };
          }
          return this._style;
        },
        configurable: true,
      });
    }
  });

  afterEach(() => {
    // 恢复原始全局变量
    global.window = originalWindow as Global['window'];
    global.document = originalDocument as Global['document'];
  });

  test('模板化', () => {
    // 创建分析器实例
    const analyzer = new DOMDomainAnalyzer({
      enableHighlight: false, // 禁用高亮以确保一致性
    });
    const analyzedHTML = analyzer.getAnalyzedHTML();

    const analyzedDOM = new JSDOM(analyzedHTML);
    const domContentExtractor = new DOMContentExtractor();
    const domContent = domContentExtractor.extract(analyzedDOM.window.document.body);

    const htmlTemplateProcessor = new HTMLTemplateProcessor();
    const processedHTML = htmlTemplateProcessor.processHTML((domContent as HTMLElement).outerHTML);

    expect(processedHTML).toMatchSnapshot();
  });
});
