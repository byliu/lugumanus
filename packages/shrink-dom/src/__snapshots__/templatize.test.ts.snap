// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`DOMDomainAnalyzer 模板化 1`] = `
"T1: <div><div><span>$1</span><span>$2</span></div><div><span>$3</span>$4</div></div>
T2: <div><div>$1<span>更多航班  </span></div><div>$2</div><div>$3<span>$4</span></div></div>
T3: <div>$1<div><div><span>$2</span><span>$3</span></div>$4</div></div>
T4: <div><div>$1</div><div><div>1 <div><div><span>$2</span><span>$3</span></div><div>$4</div></div></div><div>2 <div><div><span>$5</span><span>$6</span></div><div>$7</div></div></div><div>3 <div><div><span>$8</span><span>$9</span></div><div>$10</div></div></div><div>4 <div><div><span>$11</span><span>$12</span></div><div>$13</div></div></div><div>5 <div><div><span>$14</span><span>$15</span></div><div>$16</div></div></div></div></div>
T5: <div><div>国内 <div>国际及 中国港澳台 </div></div><div><ul>热门 ABCDEF GHIJ KLMN PQRSTUVW XYZ </ul><ul>上海 北京 成都 广州 三亚 深圳 重庆 杭州 昆明 西安 海口 哈尔滨 南京 乌鲁木齐 长沙 厦门 沈阳 武汉 天津 青岛 </ul></div></div>
T6: <div><button aria-label=$1 id=$2>$3</button><div><a id=$4>$5</a><a id=$6>$7</a></div></div>

<html><head/><body><div><div role="navigation" id="80"><div>{T6("酒店 按回车键打开菜单",90,"酒店 ",94,"国内酒店 ",96,"海外酒店 ")}<div><button aria-label="机票 按回车键打开菜单" id="99">机票 </button><div><a id="103">国内/国际/中国港澳台 </a><a id="105">特价机票 </a><a id="107">航班动态 </a><a id="109">值机选座 </a><a id="111">退票改签 </a><a id="113">报销凭证 </a><a id="115">机场攻略 </a><a id="117">定制包机 </a></div></div>{T6("火车票 按回车键打开菜单",120,"火车票 ",124,"国内火车票 ",126,"国际/中国港澳台 ")}<div><button aria-label="旅游 按回车键打开菜单" id="129">旅游 </button><div><a id="133">旅游首页 </a><a id="135">跟团游 </a><a id="137">私家团 </a><a id="139">自由行 </a><a id="141">邮轮 </a><a id="143">一日游 </a><a id="145">定制旅游 </a><a id="147">周边游 </a><a id="149">高端游 </a><a id="151">主题游 </a><a id="153">游学 </a><a id="155">签证 </a><a id="157">保险 </a><a id="159">企业会奖 </a><a id="161">目的地 </a></div></div><button aria-label="门票·活动 按回车键打开菜单" id="164">门票·活动 </button>{T6("汽车·船票 按回车键打开菜单",169,"汽车·船票 ",173,"汽车票 ",175,"船票 ")}<div><button aria-label="用车 按回车键打开菜单" id="178">用车 </button><div><a id="182">国内租车 </a><a id="184">境外租车 </a><a id="186">接送机站 </a><a id="188">按天包车 </a></div></div><button aria-label="攻略·景点 按回车键打开菜单" id="191">攻略·景点 </button><button aria-label="旅游地图 按回车键打开菜单" id="196">旅游地图 </button><div><button aria-label="全球购 按回车键打开菜单" id="202">全球购 </button><div><a id="206">名店购 </a><a id="208">银联特惠 </a><a id="210">外币兑换 </a></div></div><div><button aria-label="礼品卡 按回车键打开菜单" id="213">礼品卡 </button><div><a id="217">礼品卡首页 </a><a id="219">企业采购 </a><a id="221">心意送礼 </a><a id="223">礼品卡福袋 </a></div></div><button aria-label="携程金融 按回车键打开菜单" id="226">携程金融 </button><button aria-label="企业商旅 按回车键打开菜单" id="231">企业商旅 </button><button aria-label="老友会 按回车键打开菜单" id="236">老友会 </button><button aria-label="关于携程 按回车键打开菜单" id="241">关于携程 </button></div></div><div><div><div><a id="259">携程旅行网 </a><form><input type="text" id="262" editable="true"/><input type="text" placeholder="搜索任何旅游相关" aria-label="搜旅任何旅行相关..." id="265" editable="true"/></form></div><div><ul><button aria-label="我的账户: 欧阳w" id="272">欧阳w </button><button id="281">我的订单 </button><button aria-label="携程客服" id="287">联系客服 </button><a title="消息通知" id="289">通知 </a><button aria-label="扫码下载携程App或关注微信公众号" id="296"/></ul><button aria-label="无障碍阅读开关" id="298"/><button aria-label="关怀版" id="300"/></div></div><div><div><div><ul>国内、国际/中国港澳台 <a id="313">特价机票 </a><a id="315">航班动态 </a><a id="317">在线选座 </a><a id="319">退票改签 </a>更多服务 </ul><div><div><div><ul><span>  单程 </span><span>  往返 </span><span>  多程(含缺口程) </span><div> <div>国内机票多程适用于以下场景： <div>前往多个目的地 <div><div>去程：上海 - 北京 </div><div>返程： 天津 - 上海 </div></div></div><div>出发地和返回地不同 <div><div>去程： 上海 - 北京 </div><div>返程： 北京 - 杭州 </div></div></div></div></div></ul><div><div>不限舱等   </div><ul><li> 不限舱等 </li>经济舱 公务/头等舱 </ul></div></div><div><div><div>出发地 <div><input aria-label="可输入城市或机场" name="owDCity" type="text" placeholder="可输入城市或机场" value="成都(CTU)" id="390" editable="true"/>{T5()}</div></div><div name="exchangeBtn"> </div><div>目的地 <div><input aria-label="可输入城市或机场" name="owACity" type="text" placeholder="可输入城市或机场" value="深圳(SZX)" id="438" editable="true"/>{T5()}</div></div></div><div><div>出发日期 <div><input aria-label="请选择日期" type="text" placeholder="yyyy-mm-dd" value="2025-03-29"/>周六 明天 </div></div><a id="493">+ 添加返程 </a><div>4天 返回日期 <div><input aria-label="请选择日期" type="text" placeholder="" value="2025-04-01"/>周二 周二 </div></div></div><div>乘客类型 <div><span>带儿童 2~12岁 </span><span>带婴儿 14天-2岁 </span></div></div></div></div><button type="submit" id="521"> 搜索 </button></div></div><div><div>您搜索过的机票 <div> 清空历史 </div></div><li><div><span>成都 深圳 </span><span>查最新价  </span></div><span>03-29 去 04-01 回 </span></li></div><div><div><div>低价速报 出发地： <div><div><input aria-label="请输入出发地" name="owDCity" type="text" placeholder="请输入出发地" value="" id="580" editable="true"/> </div>{T5()}</div></div><a id="621">更多目的地  </a></div><div><div>{T4("周末省心游  ","成都  昆明 ","¥ 540 起 ",04-05 去 04-08 回 1.5折 ,"成都  贵阳 ","¥ 560 起 ",04-11 去 04-13 回 2折 ,"成都  南宁 ","¥ 600 起 ",03-29 去 04-01 回 2.2折 ,"成都  合肥 ","¥ 620 起 ",04-04 去 04-05 回 1.9折 ,"成都  长沙 ","¥ 645 起 ",04-04 去 04-05 回 2折 )}<div><div>爱上大草原  </div><div>{T3(1 ,"成都  鄂尔多斯 ","¥ 340 起 ",2025-04-23 周三 )}{T3(2 ,"成都  呼和浩特 ","¥ 390 起 ",2025-04-09 周三 )}{T3(3 ,"成都  锡林浩特 ","¥ 440 起 ",2025-04-05 周六 )}{T3(4 ,"成都  赤峰 ","¥ 510 起 ",2025-04-03 周四 )}<div>5 <div><div><span>成都  康定 </span><span>¥ 524 起 </span></div><div>2025-04-07 周一 3.4折 </div></div></div></div></div>{T4("海边浪一浪  ","成都  珠海 ","¥ 318 起 ",2025-04-14 周一 2.1折 ,"成都  深圳 ","¥ 350 起 ",2025-04-01 周二 1.4折 ,"成都  宁波 ","¥ 370 起 ",2025-05-14 周三 1.6折 ,"成都  青岛 ","¥ 395 起 ",2025-04-05 周六 1.8折 ,"成都  惠州 ","¥ 400 起 ",2025-04-12 周六 2.6折 )}{T4("美食排行榜  ","成都  西安 ","¥ 260 起 ",2025-04-08 周二 3.3折 ,"成都  长沙 ","¥ 280 起 ",2025-04-05 周六 1.6折 ,"成都  武汉 ","¥ 300 起 ",2025-04-05 周六 1.9折 ,"成都  广州 ","¥ 350 起 ",2025-04-01 周二 1.6折 ,"成都  青岛 ","¥ 395 起 ",2025-04-05 周六 1.8折 )}{T4("倾听古镇风情  ","成都  贵阳 ","¥ 260 起 ",2025-03-31 周一 1.9折 ,"成都  丽江 ","¥ 300 起 ",2025-04-06 周日 1.7折 ,"成都  桂林 ","¥ 340 起 ",2025-03-29 周六 2.7折 ,"成都  大理 ","¥ 345 起 ",2025-03-29 周六 2.1折 ,"成都  太原 ","¥ 350 起 ",2025-04-04 周五 2.6折 )}{T4("游历名山大川  ","成都  西安 ","¥ 260 起 ",2025-04-08 周二 3.3折 ,"成都  济南 ","¥ 270 起 ",2025-04-04 周五 1.1折 ,"成都  南昌 ","¥ 300 起 ",2025-04-05 周六 2.2折 ,"成都  丽江 ","¥ 300 起 ",2025-04-06 周日 1.7折 ,"成都  武夷山 ","¥ 350 起 ",2025-03-31 周一 2.3折 )}{T4("品味历史名城  ","成都  西安 ","¥ 260 起 ",2025-04-08 周二 3.3折 ,"成都  长沙 ","¥ 280 起 ",2025-04-05 周六 1.6折 ,"成都  大理 ","¥ 345 起 ",2025-03-29 周六 2.1折 ,"成都  广州 ","¥ 350 起 ",2025-04-02 周三 1.6折 ,"成都  扬州 ","¥ 400 起 ",2025-04-05 周六 2折 )}{T4("网红拍摄必选地  ","成都  三亚 ","¥ 275 起 ",2025-04-06 周日 1.1折 ,"成都  昆明 ","¥ 279 起 ",2025-05-15 周四 1.5折 ,"成都  九寨沟 ","¥ 300 起 ",2025-04-01 周二 1.9折 ,"成都  西宁 ","¥ 310 起 ",2025-04-02 周三 2.1折 ,"成都  桂林 ","¥ 340 起 ",2025-03-29 周六 2.7折 )}</div> </div></div><div><div><div>低价地图 出发地： <div><div><input aria-label="请输入出发地" name="owDCity" type="text" placeholder="请输入出发地" value="" id="1350" editable="true"/> </div>{T5()}</div></div><a id="1391">更多目的地  </a></div><div><div>一个月内最低价目的地 <div>{T2("海口 ","雷琼海口火山群世界地质公园 观澜湖华谊冯小刚电影公社(百年记忆公社) ",03-30 周日去 ,"¥ 235  ")}{T2("兰州 ","兰州水墨丹霞旅游景区 甘肃省博物馆 ",04-10 周四去 ,"¥ 240  ")}{T2("西安 ","秦始皇帝陵博物院(兵马俑) 华清宫 ",04-08 周二去 ,"¥ 260  ")}{T2("贵阳 ","青岩古镇 花溪夜郎谷 ",03-31 周一去 ,"¥ 260  ")}{T2("济南 ","趵突泉景区 千佛山 ",04-04 周五去 ,"¥ 270  ")}{T2("三亚 ","三亚亚特兰蒂斯度假区 三亚亚特兰蒂斯水世界 ",04-06 周日去 ,"¥ 275  ")}{T2("长沙 ","湖南博物院 橘子洲 ",04-05 周六去 ,"¥ 280  ")}{T2("昆明 ","石林风景区 云南民族村 ",04-09 周三去 ,"¥ 290  ")}{T2("武汉 ","黄鹤楼 晴川阁 ",04-05 周六去 ,"¥ 300  ")}{T2("南昌 ","滕王阁 南昌汉代海昏侯国遗址公园 ",04-05 周六去 ,"¥ 300  ")}</div></div><div><div><div><div>海口 <span>¥ 235 </span></div>最低价 </div><div>兰州 <span>¥ 240 </span></div><div>西安 <span>¥ 260 </span></div><div>贵阳 <span>¥ 260 </span></div><div>济南 <span>¥ 270 </span></div><div>三亚 <span>¥ 275 </span></div><div>长沙 <span>¥ 280 </span></div><div>昆明 <span>¥ 290 </span></div><div>武汉 <span>¥ 300 </span></div><div>南昌 <span>¥ 300 </span></div></div><a title="退出室内景" id="1715">出口 </a>确定 </div></div></div><div>航空公司特惠专区 <div><div> <div>中国国航 南方航空 东方航空 深圳航空 祥鹏航空 西部航空 春秋航空 海南航空 吉祥航空 天津航空 乌鲁木齐航空 北部湾航空 东海航空 九元航空 昆明航空 青岛航空 厦门航空 山东航空 首都航空 四川航空 金鹏航空 河北航空 苏南瑞丽航空 长安航空 </div> </div><div>中国国际航空股份有限公司简称“国航”，英文名称为“Air China Limited”，简称“Air China”，是中国唯一载国旗飞行的民用航空公司以及世界最大的航空联盟——星空联盟成员、2008年北京奥运会航空客运合作伙伴，具有国内航空公司第一的品牌价值(世界品牌实验室2013年评测为765.68亿元)，在… <div>查看更多 <span>中国国际航空股份有限公司简称“国航”，英文名称为“Air China Limited”，简称“Air China”，是中国唯一载国旗飞行的民用航空公司以及世界最大的航空联盟——星空联盟成员、2008年北京奥运会航空客运合作伙伴，具有国内航空公司第一的品牌价值(世界品牌实验室2013年评测为765.68亿元)，在航空客运、货运及相关服务诸方面，均处于国内领先地位。 国航的企业标识由一只艺术化的凤凰和中国改革开放的总设计师邓小平同志书写的“中国国际航空公司”以及英文“AIR CHINA”构成。国航标志是凤凰，同时又是英文“VIP”（尊贵客人）的艺术变形，颜色为中国传统的大红，具有吉祥、圆满、祥和、幸福的寓意，寄寓着国航人服务社会的真挚情怀和对安全事业的永恒追求。国航愿景和定位是“具有国际知名度的航空公司”，其内涵是实现“竞争实力世界前列、发展能力持续增强、客户体验美好独特、相关利益稳步提升”的四大战略目标；企业精神强调“爱心服务世界、创新导航未来”，企业使命是“满足顾客需求，创造共有价值”；企业价值观是“服务至高境界、公众普遍认同”；服务理念是“放心、顺心、舒心、动心”。 截至2013年12月31日，国航（含控股公司）共拥有以波音、空中客车为主的各型飞机497架，平均机龄6.33年；经营客运航线已达298条，其中国际航线71条，地区航线15条，国内航线212条，通航国家（地区）31个，通航城市154个，其中国际47个，地区3个，国内104个；通过与星空联盟成员等航空公司的合作，将服务进一步拓展到195个国家的1328个目的地。 </span></div></div><div><div>出发地： 北京 上海 天津 成都 重庆 西安 武汉 常州 广州 深圳 </div><span>{T1("北京  上海 ","￥ 395 起 ",2025-03-29 周六 ,1.9折 )}{T1("北京  成都 ","￥ 380 起 ",2025-04-05 周六 ,1.3折 )}{T1("北京  广州 ","￥ 535 起 ",2025-04-04 周五 ,1.6折 )}{T1("北京  深圳 ","￥ 574 起 ",2025-03-31 周一 ,1.6折 )}{T1("北京  重庆 ","￥ 445 起 ",2025-03-31 周一 ,1.6折 )}{T1("北京  杭州 ","￥ 350 起 ",2025-04-05 周六 ,1.1折 )}{T1("北京  昆明 ","￥ 520 起 ",2025-04-01 周二 ,1.6折 )}{T1("北京  三亚 ","￥ 495 起 ",2025-03-30 周日 ,1.2折 )}</span></div></div></div></div><div><div>| <a id="1998">行程保障 </a>| <a id="2002">价格保障 </a>| <a id="2006">7*24小时客服 </a>| <a id="2010">应急援助 </a>| <a id="2014">安心1 + 1 </a>| <a id="2018">组合交通出行 </a></div><div><div>服务公告 <div><ul><a id="2026">机票预订服务提供方：上海华程西南国际旅行社有限公司 </a><a id="2028">关于禁止携带危险品乘机的通知 </a><a id="2030">关于欺诈短信电话的提醒 </a></ul><ul>携程机票投诉受理方式 机票投诉受理热线：95010 机票投诉受理邮箱：jpwbss@trip.com </ul></div></div><div><div>携程APP 享手机订单优惠 随时随地查订单 </div><div>携程特价机票 轻松出行必备 定期放送特价 </div></div></div></div><div>携程旅行网提供机票预订，飞机票查询时刻表。票价查询以及特价机票，便宜机票，打折机票查询预订服务 <p>热门航线 <a id="2056">太原到北京机票 </a><a id="2057">上海到桂林机票 </a><a id="2058">杭州到西安机票 </a><a id="2059">杭州到广州机票 </a><a id="2060">武汉到深圳机票 </a><a id="2061">西安到深圳机票 </a><a id="2062">深圳到上海机票 </a><a id="2063">昆明到上海机票 </a><a id="2064">上海到大连机票 </a><a id="2065">上海到太原机票 </a><a id="2066">包头到北京机票 </a><a id="2067">西安到广州机票 </a><a id="2068">福州到广州机票 </a><a id="2069">广州到厦门机票 </a><a id="2070">北京到郑州机票 </a><a id="2071">哈尔滨到上海机票 </a><a id="2072">南京到北京机票 </a><a id="2073">贵阳到北京机票 </a><a id="2074">北京到上海机票 </a><a id="2075">南昌到广州机票 </a></p><p>机票工具箱 <a id="2078">打折机票 </a><a title="航班查询时刻表" id="2079">航班时刻表查询 </a><a title="航班实时起降查询" id="2080">航班动态查询 </a><a id="2081">机票地图 </a><a id="2082">值机 </a></p><p>打折机票 <a id="2085">上海特价机票 </a><a id="2086">北京特价机票 </a><a id="2087">广州特价机票 </a><a id="2088">深圳特价机票 </a><a id="2089">重庆特价机票 </a><a id="2090">成都特价机票 </a><a id="2091">昆明特价机票 </a><a id="2092">厦门特价机票 </a><a id="2093">三亚特价机票 </a><a id="2094">大连特价机票 </a><a id="2095">丽江特价机票 </a><a id="2096">天津特价机票 </a><a id="2097">福州特价机票 </a><a id="2098">沈阳特价机票 </a><a id="2099">郑州特价机票 </a><a id="2100">青岛特价机票 </a><a id="2101">西安特价机票 </a><a id="2102">南京特价机票 </a><a id="2103">武汉特价机票 </a><a id="2104">杭州特价机票 </a><a id="2105">更多特价机票 </a></p><p>航空机票 <a id="2108">春秋航空 </a><a id="2109">东方航空 </a><a id="2110">南方航空 </a><a id="2111">中国国航 </a><a id="2112">海南航空 </a><a id="2113">深圳航空 </a><a id="2114">厦门航空 </a><a id="2115">山东航空 </a><a id="2116">吉祥航空 </a><a id="2117">首都航空 </a><a id="2118">四川航空 </a><a id="2119">幸福航空 </a><a id="2120">祥鹏航空 </a><a id="2121">上海航空 </a></p></div></div><div><div><dl>旅游资讯 <a id="2132">宾馆索引 </a><a id="2134">攻略索引 </a><a id="2136">机票索引 </a><a id="2138">网站导航 </a><a id="2140">旅游索引 </a><a id="2142">邮轮索引 </a><a id="2144">企业差旅索引 </a></dl><dl>加盟合作 <a id="2149">分销联盟 </a><a id="2151">友情链接 </a><a id="2153">企业礼品卡采购 </a><a id="2155">保险代理 </a><a id="2157">代理合作 </a><a id="2159">酒店加盟 </a><a id="2161">目的地及景区合作 </a><a id="2163">更多加盟合作 </a></dl><dl>关于携程 <a id="2168">关于携程 </a>携程热点 <a id="2174">联系我们 </a><a id="2176">诚聘英才 </a><a id="2178">用户协议 </a><a id="2182">营业执照 </a><a id="2184">安全中心 </a><a id="2186">携程内容中心 </a><a id="2188">知识产权 </a><a id="2190">Trip.com Group </a></dl><div><dl><div>境内： <div>95010 或 400-830-6666 </div></div>中国香港：+852-3008-3295 中国澳门：+86-21 3406-4888 中国台湾：+86-21 3406-4888 其他国家和地区：+86-21-3406-4888 </dl>扫码下载携程App </div></div><div><p><a id="2220">沪公网备31010502002731号 </a>丨 <a id="2222">互联网药品信息服务资格证 </a>丨 <a id="2223">（沪）网械平台备字[2022]第00001号 </a>| 沪网食备1050001号 | <a id="2225">旅游度假资质 </a>| <a id="2226">平台信息 </a></p><p>违法和不良信息举报电话021-22500846丨全国旅游投诉热线12345丨 <a id="2229">上海市旅游网站落实诚信建设主体责任承诺书 </a></p><div><a title="工商亮照标识" id="2231">上海工商 </a><a title="征信网" id="2232">网络社会征信网 </a><a title="上海市互联网违法与违规信息举报中心" id="2233">信息举报中心 </a><a id="2234">可信网站 </a></div><div><a id="2236">信用中国 </a><a id="2237">网上有害信息举报专区 </a><a id="2238">消费维权联络点 </a><a id="2239">网购大家评 </a>适老化及无障碍标识 </div></div></div></div></div><input type="hidden" name="market-mate-for-1688" value="installed"/><div>同款比价 <div>暂时关闭此按钮 调整此按钮位置 </div></div></body></html>"
`;
